In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding="latin-1")

In [3]:
df

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

# Finding the total number of unique values in the Area column

In [21]:
df['Area'].nunique()

49

# Finding the shape of the dataset

In [11]:
df.shape

(60943, 12)

# Selecting columns ‘Y2017’ and ‘Area’, and  performing a groupby operation on ‘Area’ to find which of the Areas had the highest sum in 2017?

In [26]:
#Select the 'Area' and 'Y2017' columns
selected_columns = df[['Area', 'Y2017']]

#Group by 'Area' and calculate the sum for each area in 'Y2017' column
grouped_data = selected_columns.groupby('Area')['Y2017'].sum().reset_index()

#Find the area with the highest sum in 2017
area_with_highest_sum = grouped_data[grouped_data['Y2017'] == grouped_data['Y2017'].max()]

# Print the result
print("Area with the highest sum in 2017:")
print(area_with_highest_sum)

Area with the highest sum in 2017:
       Area         Y2017
34  Nigeria  1.485664e+06


# Finding the total number and percentage of missing data in 2014 to 3 decimal places

In [6]:
#Checking for the total number of missing values in 'Y2014'
total_missing = df['Y2014'].isnull().sum()

In [7]:
total_missing

1589

In [8]:
# Calculate the total number of values in the '2014' column
total_values = len(df['Y2014'])

# Calculate the percentage of missing data
percentage_missing = (total_missing / total_values) * 100

# Round the percentage to 3 decimal places
percentage_missing = round(percentage_missing, 3)

# Print the results
print(f'Total number of missing data in column "2014": {total_missing}')
print(f'Percentage of missing data in column "2014": {percentage_missing}%') 

Total number of missing data in column "2014": 1589
Percentage of missing data in column "2014": 2.607%


# Finding the total Protein supply quantity in Madagascar in 2015

In [11]:
# Filtering the DataFrame to select rows where 'Area' is 'Madagascar', 'Element' is 'Protein supply quantity (g/capita/day)', and 'Y2015' is the column of interest
filtered_df = df[(df['Area'] == 'Madagascar') & (df['Element'] == 'Protein supply quantity (g/capita/day)')]

# Extracting the '2015' column from the filtered DataFrame and calculate the total quantity
total_protein_supply_2015 = filtered_df['Y2015'].sum()

print(f'Total Protein supply quantity in Madagascar in 2015: {total_protein_supply_2015}')

Total Protein supply quantity in Madagascar in 2015: 173.04999999999998


# Performing a groupby operation on ‘Element’ and finding the total number of the sum of Processing in 2017

In [13]:
# Grouping the DataFrame by the 'Element' column and calculating the sum of '2017' for each unique 'Element'
grouped = df.groupby('Element')['Y2017'].sum()

# Selecting the sum corresponding to 'Processing'
total_processing_2017 = grouped['Processing']

print(f'Total sum of Processing in 2017: {total_processing_2017}')


Total sum of Processing in 2017: 292836.0


# Finding the year that had the least correlation with ‘Element Code’

In [20]:
df.corr()['Element Code']

C:\Users\Administrator\AppData\Local\Temp\ipykernel_10352\1992242030.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()['Element Code']


Area Code      -0.000209
Item Code      -0.024683
Element Code    1.000000
Y2014           0.024457
Y2015           0.023889
Y2016           0.023444
Y2017           0.024254
Y2018           0.024279
Name: Element Code, dtype: float64

From the output above, the year with the least correlation with "Element Code" is Y2016 with a correlation of 0.023444

# Performing a groupby operation on ‘Element’ and finding the year has the highest sum of Stock Variation?

In [33]:
# Filtering the DataFrame to select rows where 'Element' is 'Stock Variation'
stock_variation_df = df[df['Element'] == 'Stock Variation']

# Grouping by 'Element' and calculate the sum for each year
grouped_data = stock_variation_df.groupby('Element')[['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']].sum()

print(f"Sum of Stock Variation in all years:\n{grouped_data}")

Sum of Stock Variation in all years:
                    Y2014     Y2015     Y2016     Y2017     Y2018
Element                                                          
Stock Variation  58749.83  34910.99  33140.12  54316.91  20577.91


In [32]:
# Finding the year with the highest sum of Stock Variation
highest_sum_year = grouped_data.idxmax(axis=1)
print(f"The year with the highest sum of Stock Variation is: {highest_sum_year[0]}")

The year with the highest sum of Stock Variation is: Y2014


In [42]:
my_tuppy = (1,2,5,8)

my_tuppy[2] = 6

TypeError: 'tuple' object does not support item assignment

# Selecting columns ‘Y2017’ and ‘Area’, performing a groupby operation on ‘Area’ and finding which of the Areas had the 7th lowest sum in 2017

In [37]:
# Selecting columns 'Y2017' and 'Area'
selected_columns = df[['Y2017', 'Area']]

# Grouping by 'Area' and calculating the sum for each area in 'Y2017' column
grouped_data = selected_columns.groupby('Area')['Y2017'].sum().reset_index()

# Sorting the data by the sum in ascending order
sorted_data = grouped_data.sort_values(by='Y2017')

# Finding the Area with the 7th lowest sum in 2017
seventh_lowest_area = sorted_data.iloc[6]['Area']

print(f"The area with the 7th lowest sum in 2017 is: {seventh_lowest_area}")

The area with the 7th lowest sum in 2017 is: Guinea-Bissau


# Finding the mean and standard deviation across the whole dataset for the year 2017 to 2 decimal places

In [39]:
# Select the 'Y2017' column from the DataFrame
y2017_column = df['Y2017']

# Calculate the mean and standard deviation for the 'Y2017' column
mean_2017 = round(y2017_column.mean(), 2)
std_deviation_2017 = round(y2017_column.std(), 2)

print(f"Mean for the year 2017: {mean_2017}")
print(f"Standard Deviation for the year 2017: {std_deviation_2017}")

Mean for the year 2017: 140.92
Standard Deviation for the year 2017: 1671.86


In [43]:
S = [['him', 'sell'], [90, 28, 43]]
S[0][1][1]

'e'

# Performing a groupby sum aggregation on ‘Item’ to find the total sum of Wine produced in 2015 and 2018

In [41]:
# Filter the DataFrame to select rows where 'Item' is 'Wine'
wine_df = df[df['Item'] == 'Wine']

# Calculate the total sum of Wine produced in 2015 and 2018
total_wine_2015 = wine_df['Y2015'].sum()
total_wine_2018 = wine_df['Y2018'].sum()

print(f"Total Wine produced in 2015: {total_wine_2015}")
print(f"Total Wine produced in 2018: {total_wine_2018}")

Total Wine produced in 2015: 4251.8099999999995
Total Wine produced in 2018: 4039.3199999999997


In [44]:
y = [(2, 4), (7, 8), (1, 5, 9)]

In [45]:
x = y[1][1]
print(x)

8


In [47]:
lst = [[35, 'Portugal', 94], [33, 'Argentina', 93], [30 , 'Brazil', 92]]
col = ['Age','Nationality','Overall']
pd.DataFrame(lst, columns = col, index = [i for i in range(1,4)])

Age Nationality  Overall
1   35    Portugal       94
2   33   Argentina       93
3   30      Brazil       92